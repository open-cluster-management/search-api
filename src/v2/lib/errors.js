/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright (c) 2021 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import { createError } from 'apollo-errors';

export const GenericError = createError('GenericError', {
  message: 'A generic error has occurred', // TODO: NLS
});

export const NetworkError = createError('NetworkError', {
  message: 'A network error has occurred', // TODO: NLS
});

export const AuthenticationError = createError('AuthenticationError', {
  message: 'An AuthenticationError error has occurred', // TODO: NLS
});
