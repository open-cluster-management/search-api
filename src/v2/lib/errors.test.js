/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright Contributors to the Open Cluster Management project

import { AuthenticationError, GenericError, NetworkError } from './errors';

describe('Errors', () => {
  test('Test Authentication Error', async () => {
    expect(AuthenticationError).toMatchSnapshot();
  });
  test('Test Generic Error', async () => {
    expect(GenericError).toMatchSnapshot();
  });
  test('Test Network Error', async () => {
    expect(NetworkError).toMatchSnapshot();
  });
});
