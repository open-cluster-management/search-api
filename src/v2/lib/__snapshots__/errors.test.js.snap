// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Authentication Error 1`] = `[Function]`;

exports[`Errors Generic Error 1`] = `[Function]`;

exports[`Errors Network Error 1`] = `[Function]`;
